### DISCUSSION TITLE
🚀 Feature Request: Build & Deploy Cloudflare Snippets

### DISCUSSION THREAD

**Discussion by nathanclevenger** [cloudflare/workers-sdk#13485] [Feature Requests]
### Describe the solution

It would be great if there is a way to use wrangler to build & deploy a code file to Cloudflare Snippets so that snippets can be kept in source control.

Today, to keep snippets code in source control you have to do something like `npx wrangler deploy snippet.ts --dry-run --outdir dist` and then something along the lines of:
```
curl --request PUT https://api.cloudflare.com/client/v4/zones/{zone_id}/snippets/{snippet_name} \
--header "Authorization: Bearer <API_TOKEN>" \
--header "Content-Type: multipart/form-data" \
--form files=@worker.js \
--form metadata='{"main_module":"worker.js"}'
```
or copy/paste the `worker.js` into a snippet via the dashboard

It would be great if there was a simpler integration via wrangler.  Is support for snippets at all planned?

**Comment by sidharthachatterjee** [comment]
Just hit a use case that called for a Snippet and yes, this would be nice to have, in my opinion

**Comment by tstm** [comment]
@penalosa Is there some timeline on this perhaps? It'd be useful to be able to use snippets as well as workers through a similar pipeline.

**Comment by petebacondarwin** [comment]
There are currently no plans to support snippets deployment via Wrangler.

**Reply by jamestalmage** [reply]
> There are currently no plans to support snippets deployment via Wrangler.

@petebacondarwin Why not? Is it a team bandwidth issue, or some other reason it's not seen as a fit?
